server/src/main/java/org/apache/gravitino/server/web/rest/PermissionOperations.java

```diff
@@ -65,11 +69,19 @@ public Response grantRolesToUser(
       return Utils.doAs(
           httpRequest,
           () ->
-              Utils.ok(
-                  new UserResponse(
-                      DTOConverters.toDTO(
-                          accessControlManager.grantRolesToUser(
-                              metalake, request.getRoleNames(), user)))));
+              TreeLockUtils.doWithTreeLock(
+                  NameIdentifier.of(AuthorizationUtils.ofUserNamespace(metalake).levels()),
```

Comment:
I think the behavior is to update the user with new roles, so user-level write lock is enough, no need to have a write lock for the user namespace, WDYT?

Reply:
Yes, it's enough for this case. I just want to follow the other lock style. It's a little weird that some write operations use parent write lock and some write operations use self write lock.  
It's ok to modify here.

Reply:
Done.